src/commands/execute.ts

```diff
@@ -0,0 +1,199 @@
+
+export default class Execute extends Command {
+  // TODO: add description
+  public static description =
+    'This commands executes created integration. Commands `prepare`, `new` and `map` must be run before this command.';
+
+  public static examples = [
+    'superface execute <provider-name> <optional-profile-scope>.<profile-name> <optional-language>',
+  ];
+
+  public static args = [
+    {
+      name: 'providerName',
+      description: 'Name of provider.',
+      required: true,
+    },
+    {
+      name: 'profileId',
+      description: 'Id of profile, eg: starwars.character-information',
+      required: true,
+    },
+    {
+      name: 'language',
+      description: 'Language of generated integration. Default is JS.',
+      required: false,
+      default: 'JS',
+    },
+  ];
+
+  public static flags = {
+    ...Command.flags,
+  };
+
+  public async run(): Promise<void> {
+    const { flags, args } = this.parse(Execute);
+    await super.initialize(flags);
+    await this.execute({
+      logger: this.logger,
+      userError: this.userError,
+      flags,
+      args,
+    });
+  }
+
+  public async execute({
+    logger,
+    userError,
+    // flags,
+    args,
+  }: {
+    logger: ILogger;
+    userError: UserError;
+    flags: Flags<typeof Execute.flags>;
+    args: { providerName?: string; profileId?: string; language?: string };
+  }): Promise<void> {
+    const { providerName, profileId, language } = args;
+
+    // TODO: resuse check from New command
+    // Check provider name
+    if (providerName === undefined) {
+      throw userError(
+        'Missing provider name. Please provide it as first argument.',
+        1
+      );
+    }
+
+    if (!isValidProviderName(providerName)) {
+      throw userError('Invalid provider name', 1);
+    }
+
+    if (!(await exists(buildProviderPath(providerName)))) {
+      throw userError(
+        `Provider ${providerName} does not exist. Make sure to run "sf prepare" before running this command.`,
+        1
+      );
+    }
+
+    const providerJsonFile = await readFile(
+      buildProviderPath(providerName),
+      'utf-8'
+    );
+
+    let providerJson: ProviderJson;
+    try {
+      providerJson = JSON.parse(providerJsonFile) as ProviderJson;
+    } catch (e) {
+      throw userError(`Invalid provider.json file.`, 1);
+    }
+
+    try {
+      assertProviderJson(providerJson);
+    } catch (e) {
+      if (e instanceof AssertionError) {
+        throw userError(`Invalid provider.json file. ${e.message}`, 1);
+      }
+      throw userError(`Invalid provider.json file.`, 1);
+    }
+
+    if (providerName !== providerJson.name) {
+      throw userError(
+        `Provider name in provider.json file does not match provider name in command.`,
+        1
+      );
+    }
+
+    // Check profile name
+    if (profileId === undefined) {
+      throw userError(
+        'Missing profile id. Please provide it as first argument.',
```

Comment:
issue: provider is first argument, this one is second, and as in case of the provider arg this one is also marked as required